src/linker/Linker/TypeReferenceExtensions.cs

```diff
@@ -11,8 +11,11 @@ public static class TypeReferenceExtensions
 		public static string GetDisplayName (this TypeReference type)
 		{
 			var builder = GetDisplayNameWithoutNamespace (type);
-			builder.Insert (0, ".");
-			builder.Insert (0, type.GetNamespaceDisplayName ());
+			var namespaceDisplayName = type.GetNamespaceDisplayName ();
```

Comment:
nit: I'd prefer to use Append over Delete

Reply:
Yeah -- I'd also prefer using append in the whole `GetDisplayName` logic, I remember trying to use append instead but (IIRC) it got difficult in presence generics. I opened https://github.com/mono/linker/issues/1975.